### ISSUE TITLE
Refactor names which conflict with Rust keywords

### ISSUE THREAD

**Issue by Michael-Lfx** [gfx-rs/gfx#2525]
For example `image::ViewKind` reflects to `VkImageViewCreateInfo::viewType` actually. It was renamed to `kind` because it conflicts with keyword `type`. Now we can use `r#*` like `r#type`. Maybe renaming they back to what they should be will results in a better understanding.

**Comment by omni-viral** [comment]
IMHO
`r#keyword` could be used as quick-fix for conflict with newly added keyword to stay compatible with older version.
But using it to reflect names from `Vulkan` which `gfx-hal` mimics but not 1:1 is not so great.

**Comment by kvark** [comment]
Yeah, I don't think we should consider `r#` unless there is no choice.
